src/Http/Http/src/Internal/StreamResponseBodyFeature.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.AspNetCore.Http
+{
+    public class StreamResponseBodyFeature : IHttpResponseBodyFeature
+    {
+        private PipeWriter _pipeWriter;
+        private bool _started;
+
+        public StreamResponseBodyFeature(Stream stream)
+        {
+            Body = stream ?? throw new ArgumentNullException(nameof(stream));
+        }
+
+        public Stream Body { get; }
+
+        public PipeWriter Writer
+        {
+            get
+            {
+                if (_pipeWriter == null)
+                {
+                    _pipeWriter = PipeWriter.Create(Body, new StreamPipeWriterOptions(leaveOpen: true));
+                }
+
+                return _pipeWriter;
+            }
+        }
+
+        public virtual void DisableResponseBuffering()
+        {
+        }
+
+        public virtual async Task SendFileAsync(string path, long offset, long? count, CancellationToken cancellation)
+        {
+            if (!_started)
+            {
+                await StartAsync(cancellation);
+            }
+            await SendFileAsyncCore(Body, path, offset, count, cancellation);
+        }
+
+        public virtual Task StartAsync(CancellationToken token = default)
+        {
+            if (!_started)
+            {
+                _started = true;
+                return Body.FlushAsync(token);
+            }
+            return Task.CompletedTask;
+        }
+
+        public virtual async Task CompleteAsync()
+        {
+            if (!_started)
+            {
+                await StartAsync();
+            }
```

Comment:
```suggestion
            }

```

Normally, I'd say this was a nit, but I really though this was an "if else" for a second.

Reply:
I mean it's still a nit, but not the nittiest 😄